README.md
# responsive_ui

A new Flutter package for android ,ios which help users toy make the ui ,ore responsive.



## Screens
<image src="1.png" height="300em" />
<image src="2.png" height="300em" />
<image src="3.png" height="300em" />
<image src="4.png" height="300em" />
<image src="5.png" height="300em" />
<image src="6.png" height="300em" />
<image src="7.png" height="300em" />




### How To Use It




["Example"](https://github.com/momagdy194/responsive_ui/blob/main/example/example.dart)


*add the dependencies to your file.

```yaml
  dependencies:
    flutter:
        sdk:
  responsive_ui:
```


```dart



/// i have make thats class you will wrap your widget with "ResponsiveUi" class which he cant tack parameter 
///  it is type is widget and make inistance form Defice info its name is size 
///  it will give me type of widget size 
class MyClass extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(body: 
    
    
    ResponsiveUi(builder: (context, size)=> /// that is my class i wrap my widget with it 
    
    ListView(children: [
/// and you can  mange  the size in your widget with replace the 
/// size.screenWidth or ech like that by  size.screenWidth or size.screenHeight
///and you can replace 
      Text('title',style: TextStyle(fontSize: size.screenWidth)),
      Text('title',style: TextStyle(fontSize: size.screenHeight)),
      Text('title',style: TextStyle(fontSize: size.titleText)),///titleText
      Text('subTitle',style: TextStyle(fontSize: size.normalText))///normal

      ,Icon(Icons.error,size: size.normalIconSize)/// normal iconSize;
      ,Icon(Icons.error,size: size.mediumIconSize,)///medium iconSize;
      ,Icon(Icons.error,size: size.bigIconSize,)///big iconSize;
      ,
      Container(height:size.normalButton,child: RaisedButton(onPressed: (){},child: Text("Normal Button",)))///normalButton
      ,SizedBox(height: 15,)
      ,  Container(height:size.largeButton,child: RaisedButton(onPressed: (){},child: Text("Large Button",)))///largeButton


      ,Image(image: AssetImage("assets/images/userImage.png"),height:size.largeImage ,),///large image,
      Image(image: AssetImage("assets/images/userImage.png"),height:size.mediumImage ,),///medium image,
      Image(image: AssetImage("assets/images/userImage.png"),height:size.smallImage  ,),///small image,


      SizedBox(height: 19,)
    ],) ,));
  }
}
```








Responsive Framework adapts your UI to different screen sizes automatically. Create your UI once and have it display pixel perfect on mobile, tablet, and desktop!

[responsive_ui](https://github.com/momagdy194/responsive_ui),
a library module containing code that can be shared easily across
multiple Flutter or Dart projects.

For help getting started with Flutter, view our 
[online documentation](https://github.com/momagdy194/responsive_ui/blob/main/README.md), which offers tutorials,
samples, guidance on mobile development, and a full API reference.


CHANGELOG.md
## [0.0.1] - TODO: ReleasZ.

* TODO: Describe initial release.


test/responsive_ui_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:responsive_ui/responsive_ui.dart';



lib/responsive_ui.dart
library responsive_ui;

import 'package:flutter/material.dart';
import 'package:responsive_ui/src/finctions/get_device_type.dart';
import 'package:responsive_ui/src/models/device_info_model.dart';


class ResponsiveUi extends StatelessWidget {
  final Widget Function(BuildContext context, DeviceInfo size) builder;

  const ResponsiveUi({Key key, this.builder}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return LayoutBuilder(
      builder: (context, constrains) {
        var mediaQueryData = MediaQuery.of(context);
        var size = DeviceInfo(
          orientation: mediaQueryData.orientation,
          deviceType: getDeviceType(mediaQueryData),
          screenWidth: mediaQueryData.size.width,
          screenHeight: mediaQueryData.size.height,
          normalText: mediaQueryData.size.width * .04,
          titleText: mediaQueryData.size.width * .067,
          normalIconSize: mediaQueryData.size.width * .08,
          mediumIconSize: mediaQueryData.size.width * .12,
          bigIconSize: mediaQueryData.size.width * .15,
          backGroundIconSize: mediaQueryData.size.width * .22,
          largeImage: mediaQueryData.size.width / 1.5,
          mediumImage: mediaQueryData.size.width / 2,
          smallImage: mediaQueryData.size.width / 4,
          normalButton: mediaQueryData.size.width * .066,
          largeButton: mediaQueryData.size.width * .09,
        );
        return builder(context, size);
      },
    );
  }
}


lib/src/models/device_info_model.dart

import 'package:flutter/material.dart';
import 'package:responsive_ui/src/enums/enums.dart';

class DeviceInfo {
  final Orientation orientation;
  final DeviceType deviceType;
  final double screenWidth;
  final double screenHeight;
  final double localWidth;
  final double localHeight;
  final double normalText;
  final double titleText;
  final double normalIconSize;
  final double mediumIconSize;
  final double bigIconSize;
  final double backGroundIconSize;
  final double largeImage;
  final double mediumImage;
  final double smallImage;
  final double normalButton;
  final double largeButton;

  DeviceInfo({
    this.orientation,
    this.deviceType,
    this.screenWidth,
    this.screenHeight,
    this.localWidth,
    this.localHeight,
    this.normalText,
    this.titleText,
    this.normalIconSize,
    this.mediumIconSize,
    this.bigIconSize,
    this.largeImage,
    this.mediumImage,
    this.smallImage,
    this.normalButton,
    this.largeButton,
    this.backGroundIconSize,
  });
}


example/example.dart
import 'package:flutter/material.dart';
import 'package:responsive_ui/responsive_ui.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: MyClass(),
    );
  }
}


class MyClass extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(body: ResponsiveUi(builder: (context, size) =>ListView(children: [

      Text('title',style: TextStyle(fontSize: size.titleText)),///titleText
      Text('subTitle',style: TextStyle(fontSize: size.normalText))///normal

      ,Icon(Icons.error,size: size.normalIconSize)/// normal iconSize;
      ,Icon(Icons.error,size: size.mediumIconSize,)///medium iconSize;
      ,Icon(Icons.error,size: size.bigIconSize,)///big iconSize;
      ,
      Container(height:size.normalButton,child: RaisedButton(onPressed: (){},child: Text("Normal Button",)))///normalButtotn
      ,SizedBox(height: 15,)
      ,  Container(height:size.largeButton,child: RaisedButton(onPressed: (){},child: Text("large Button",)))///largeButton
      ///
      ,Image(image: AssetImage("assets/images/userImage.png"),height:size.largeImage ,),///large image,
      Image(image: AssetImage("assets/images/userImage.png"),height:size.mediumImage ,),///medium image,
      Image(image: AssetImage("assets/images/userImage.png"),height:size.smallImage  ,),///small image,


      SizedBox(height: 19,)
    ],) ,));
  }
}
